system/lib/wasmfs/js_api.cpp

```diff
@@ -36,7 +42,13 @@ void* _wasmfs_read_file(char* path) {
   // first 8 bytes. The remaining bytes will contain the buffer contents. This
   // allows the caller to use HEAPU8.subarray(buf + 8, buf + 8 + length).
   off_t size = file.st_size;
-  uint8_t* result = (uint8_t*)malloc(size + sizeof(size));
+  static thread_local std::unique_ptr<std::vector<char>> allocation;
```

Comment:
I don't think you need to use `std::unique_ptr`.. can't use just do `static thread_local std::vector<char> allocation`?